dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java

```diff
@@ -502,4 +502,33 @@ void deleteOrganisationUnit( OrganisationUnit organisationUnit )
      * Update all OUs (thus forcing update of path).
      */
     void forceUpdatePaths();
+
+    /**
+     * Get the count of the organisation units in the capture scope of current
+     * user
+     *
+     * @return the count of the organisation units in capture scope of current
+     *         user
+     */
+    int getCaptureOrganisationUnitCount();
+
+    /**
+     * Get the full list of organisation unit uids that falls under the capture
+     * scope of the current user
+     *
+     * @return list of org unit uids in capture scope of current user.
+     */
+    List<String> getCaptureOrganisationUnitUidsWithChildren();
+
+    /**
+     * Cached version of {@link #getCaptureOrganisationUnitCount()
+     * getCaptureOrganisationUnitCount} method If not in cache it populates it
+     * using the {@link #getCaptureOrganisationUnitCount()
+     * getCaptureOrganisationUnitCount} method
+     *
+     * @return the count of the organisation units in capture scope of current
+     *         user
+     */
+    int getCaptureOrganisationUnitCountCached();
```

Comment:
Why do we have 2 separate methods for cached and not cached service call?
In my opinion the cache should be transparent to the client of the service

Reply:
I tried to follow the same approach as used in other method in the same class here https://github.com/dhis2/dhis2-core/blob/100ef730eb8eb30254935dbf61de9147275a6c5c/dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java#L388

The client could decide if it needs realtime values or okay with "maybe" stale values. but its arguable I agree.

Reply:
Ok, if it is something that is done already and this is a temporary fix, I think is not the time to argue about this

Reply:
I have changed it slightly to make it transparent. Also included stians comments.